includes/Checker/Check_Runner.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Interface WordPress\Plugin_Check\Checker\Check_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for runner classes like AJAX runner or CLI runner.
+ *
+ * @since n.e.x.t
+ */
+interface Check_Runner {
+
+	/**
+	 * Determine if the current request is intended for the plugin checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean Returns true if the check is for plugin else false.
+	 */
+	public function is_plugin_check();
+
+	/**
+	 * Handles running the universal preparations depending on the requested checks.
```

Comment:
This is a bit too specific, the method is simply to prepare the request, and _how_ that happens depends on the implementation. Also, it's missing the `@since` annotation.
```suggestion
	 * Prepares the environment for running the requested checks.
	 *
	 * @since n.e.x.t
```